### ISSUE TITLE
Protected branches API doesn't allow individual users

### ISSUE THREAD

**Issue by GJKrupa** [gitlab4j/gitlab4j-api#479]
The API supports passing an array of both individual users and access levels for control of protected branches.  This is not supported by GitLab4J.  For example:
```
{"protected_branch":{"merge_access_levels_attributes":[{"id":100,"access_level":40},{"user_id":23}],"push_access_levels_attributes":[{"id":99,"access_level":40}]}}
```

**Comment by gmessner** [comment]
@GJKrupa 
Release 4.14.2 is now available which adds support  to the ```ProtectedBranchesApi``` to set the allowed users and groups, see the following methods in ```ProtectedBranchesApi``` :
```
// To set access by user ID
 public ProtectedBranch protectBranch(Object projectIdOrPath, String branchName,
	    Integer allowedToPush, Integer allowedToMerge, Integer allowedToUnprotect,
	    Boolean codeOwnerApprovalRequired)

// To set any of access_level, user_id, or group_id
public ProtectedBranch protectBranch(Object projectIdOrPath, String branchName,
	    AllowedTo allowedToPush, AllowedTo allowedToMerge, AllowedTo allowedToUnprotect,
	    Boolean codeOwnerApprovalRequired)
```
I am not able to test this functionality because it requires GitLab Starter, Bronze, or higher.  Please verify and close this issue when you have time.

**Comment by GJKrupa** [comment]
Looks good to me.  I've used the following code
```
        val me = gitLabApi.userApi.currentUser.id
        gitLabApi.projectApi.addMember(project.id, me, AccessLevel.MAINTAINER)
        gitLabApi.protectedBranchesApi.unprotectBranch(project.id, "master")
        gitLabApi.protectedBranchesApi.protectBranch(
            project.id,
            "master",
            AllowedTo(null, me, null),
            AllowedTo(null, me, null),
            AllowedTo(null, me, null),
            false
        )
```

and the result is as per the attached image:

<img width="1007" alt="Screenshot_2019-11-29 10 54 15_eNZRmD" src="https://user-images.githubusercontent.com/5756726/69864415-c6ec7100-1296-11ea-970c-b721a3ceae9d.png">

**Comment by GJKrupa** [comment]
Looks like there might be a glitch.  If I change the code to:
```
        gitLabApi.protectedBranchesApi.protectBranch(
            project.id,
            "master",
            me,
            me,
            me,
            false
        )
```

Then I'm seeing the following:

<img width="983" alt="Screenshot_2019-11-29 11 31 00_IYuRQh" src="https://user-images.githubusercontent.com/5756726/69866357-c0142d00-129b-11ea-84b1-b0c700f080f1.png">

**Comment by gmessner** [comment]
@GJKrupa 
I found a bug in the ```protectBranch()``` method that took user IDs.  This has been fixed and a release will be done shortly.

**Comment by gmessner** [comment]
@GJKrupa 
I just published the 4.14.4 release with this fix.  Please verify when you have a chance.  When you do verify I would suggest that you unprotect the branch first so you are testing from a known state.

**Comment by GJKrupa** [comment]
Looks good to me, thanks (the test is running in a new repo each time so always working from a clean state).
